keps/sig-scheduling/3094-pod-topology-spread-considering-taints/README.md

```diff
@@ -0,0 +1,861 @@
+only system-generated unschedulable taints
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Refactor or introduce any breaking changes to PodTopologySpread plugin
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+A new field `TaintStrategy` defined in `TopologySpreadConstraint` will be available for
+end-users to specify different strategies about how to treat taints in computing skew.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+When computing pod topology spread skew, I want to exclude nodes that don't tolerate
+all taints to prevent pods from falling into unexpected Pending state. I can simply
+configure this by specifying a taintStrategy:
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+    name: mypod
+    spec:
+      topologySpreadConstraints:
+        - maxSkew: 1
+          topologyKey: zone
+          whenUnsatisfiable: DoNotSchedule
+          labelSelector:
+            matchLabels:
+              foo: bar
+          taintStrategy: "honorAllTaints"
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- Looping all the toleration/taints to filter out unsatisfied nodes may
+lead to performance problem.
+- Some usual schedulable nodes may become unschedulable, which might be
+confusing for end-users.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+A new field named `TaintStrategy` will be introduced to `TopologySpreadConstraint`:
```

Comment:
Proposed a similar idea in https://github.com/kubernetes/kubernetes/issues/106127#issuecomment-1006207650. Let's get a consensus on how to express its semantics explicitly.

My idea is to add an extendable field that is not limited to "taints".